subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/ActionAwareConsumerConnection.java

```diff
@@ -1,86 +0,0 @@
-/*
- * Copyright 2013 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.tooling.internal.consumer.connection;
-
-import org.gradle.tooling.BuildAction;
-import org.gradle.tooling.BuildActionFailureException;
-import org.gradle.tooling.internal.adapter.ProtocolToModelAdapter;
-import org.gradle.tooling.internal.consumer.parameters.ConsumerOperationParameters;
-import org.gradle.tooling.internal.consumer.versioning.ModelMapping;
-import org.gradle.tooling.internal.consumer.versioning.VersionDetails;
-import org.gradle.tooling.internal.protocol.BuildResult;
-import org.gradle.tooling.internal.protocol.ConnectionVersion4;
-import org.gradle.tooling.internal.protocol.InternalBuildActionExecutor;
-import org.gradle.tooling.internal.protocol.InternalBuildActionFailureException;
-import org.gradle.tooling.internal.protocol.ModelBuilder;
-import org.gradle.tooling.model.gradle.BuildInvocations;
-
-import java.io.File;
-
-/**
- * An adapter for {@link InternalBuildActionExecutor}.
- *
- * <p>Used for providers >= 1.8 and <= 2.0</p>
- */
-public class ActionAwareConsumerConnection extends AbstractPost12ConsumerConnection {
```

Comment:
This needs to stay. It's part of the cross version protocol between the TAPI consumer and the Gradle provider. Same for all of the types under `org.gradle.tooling.internal.protocol`.